src/app/projects/components/project-settings/components/data-schema/data-schema.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { attributeVisibilityStates, getTooltipVisibilityState } from '../../../create-new-attribute/attributes-visibility-helper';
+import { Attribute, AttributeVisibilityStates } from '../../entities/attribute.type';
+
+
+@Component({
+  selector: 'kern-data-schema',
+  templateUrl: './data-schema.component.html',
+  styleUrls: ['./data-schema.component.scss']
+})
+export class DataSchemaComponent implements OnChanges {
+
+  @Input() project: Project;
+  @Input() attributeVisibilityStates: AttributeVisibilityStates;
+  @Input() attributes: Attribute[];
+  @Input() pKeyValid: boolean;
+
+  tooltipsArray: string[] = [];
+
+  constructor(private projectApolloService: ProjectApolloService) {
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.attributes = changes.attributes?.currentValue;
+    this.pKeyValid = changes.pKeyValid?.currentValue;
+    this.attributes.forEach((attribute) => {
+      this.tooltipsArray.push(getTooltipVisibilityState(attribute.visibility));
+    });
```

Comment:
since this.tooltipsArray isn't initialized to empty this will result in an ever-growing array (probably also the reason the warning was disabled in the dropdown component?)